src/storage.rs

```diff
@@ -669,18 +689,23 @@ mod test {
         let mut conf_state = ConfState::default();
         conf_state.nodes = nodes.clone();
 
+        let unavailable = Err(RaftError::Store(
+            StorageError::SnapshotTemporarilyUnavailable,
+        ));
         let mut tests = vec![
-            (4, Ok(new_snapshot(4, 4, nodes.clone()))),
-            (5, Ok(new_snapshot(5, 5, nodes.clone()))),
+            (4, Ok(new_snapshot(4, 4, nodes.clone())), 0),
+            (5, Ok(new_snapshot(5, 5, nodes.clone())), 5),
+            (5, Ok(new_snapshot(6, 5, nodes.clone())), 6),
+            (5, unavailable, 6),
         ];
-        for (i, (idx, wresult)) in tests.drain(..).enumerate() {
+        for (i, (idx, wresult, windex)) in tests.drain(..).enumerate() {
             let storage = MemStorage::new();
             storage.wl().entries = ents.clone();
             storage.wl().raft_state.hard_state.commit = idx;
             storage.wl().raft_state.hard_state.term = idx;
             storage.wl().raft_state.conf_state = conf_state.clone();
 
-            let result = storage.snapshot();
+            let result = storage.snapshot(windex);
```

Comment:
We should trigger unavilable if `wresult` is an error.

Reply:
Could you explain it with more details? I thinks it's already what we want.